server/src/services/library.service.ts

```diff
@@ -465,63 +442,145 @@ export class LibraryService extends BaseService {
         },
       })),
     );
+
     return JobStatus.SUCCESS;
   }
 
-  @OnJob({ name: JobName.LIBRARY_SYNC_ASSET, queue: QueueName.LIBRARY })
-  async handleSyncAsset(job: JobOf<JobName.LIBRARY_SYNC_ASSET>): Promise<JobStatus> {
-    const asset = await this.assetRepository.getById(job.id);
-    if (!asset) {
-      return JobStatus.SKIPPED;
-    }
+  @OnJob({ name: JobName.LIBRARY_SYNC_ASSETS, queue: QueueName.LIBRARY })
+  async handleSyncAssets(job: JobOf<JobName.LIBRARY_SYNC_ASSETS>): Promise<JobStatus> {
+    const assets = await this.assetRepository.getByIds(job.assetIds);
 
-    const markOffline = async (explanation: string) => {
-      if (!asset.isOffline) {
-        this.logger.debug(`${explanation}, removing: ${asset.originalPath}`);
-        await this.assetRepository.updateAll([asset.id], { isOffline: true, deletedAt: new Date() });
+    const assetIdsToOffline: string[] = [];
+    const assetIdsToOnline: string[] = [];
+    const assetIdsToUpdate: string[] = [];
+
+    this.logger.debug(`Checking batch of ${assets.length} existing asset(s) in library ${job.libraryId}`);
+
+    for (const asset of assets) {
+      const action = await this.checkExistingAsset(asset, job.libraryId);
+      switch (action) {
+        case AssetSyncResult.OFFLINE: {
+          assetIdsToOffline.push(asset.id);
+          break;
+        }
+        case AssetSyncResult.UPDATE: {
+          assetIdsToUpdate.push(asset.id);
+          break;
+        }
+        case AssetSyncResult.CHECK_OFFLINE: {
+          const isInImportPath = job.importPaths.find((path) => asset.originalPath.startsWith(path));
+
+          if (isInImportPath) {
+            const isExcluded = job.exclusionPatterns.some((pattern) => picomatch.isMatch(asset.originalPath, pattern));
+
+            if (isExcluded) {
+              this.logger.verbose(
+                `Offline asset ${asset.originalPath} is in an import path but still covered by exclusion pattern, keeping offline in library ${job.libraryId}`,
+              );
+            } else {
+              this.logger.debug(`Offline asset ${asset.originalPath} is now online in library ${job.libraryId}`);
+              assetIdsToOnline.push(asset.id);
+            }
```

Comment:
Here you don't need the else case either if you just `break` out of the case early.